pkg/ruler/compat.go

```diff
@@ -80,6 +80,10 @@ func (a *PusherAppender) AppendHistogram(_ storage.SeriesRef, l labels.Labels, t
 	return 0, nil
 }
 
+func (a *PusherAppender) AppendCTZeroSample(_ storage.SeriesRef, _ labels.Labels, _, _ int64) (storage.SeriesRef, error) {
```

Comment:
Note to reviewers: this will not get called in our use case (also the Prometheus feature is currently under an experimental feature flag, disabled by default).

Reply:
what's the prometheus ingestion behaviour when these samples are sent and the flag is disabled?

I'm afraid that we start returning 5xx errors and lead the client into an infinite loop of retries because mimir doesn't support these reset samples.

Reply:
I also feel like we should document this behaviour. Perhaps in the HTTP API docs next to the prometheus-compatibility docs.

Reply:
I thought that we would reject the created timestamp samples in remote-write requests. But actually this change is for the appender that the ruler uses. The ruler shouldn't produce these. And created timestamps aren't included in remote-write requests from prometheus yet ([code](https://github.com/prometheus/prometheus/pull/12733/files#diff-c3e3e58bc7cc8d30994e13653e830dc63e3fbac46a26f2afb9a9f65fbf158b0d))